tests/integration/http/http_sink_test.go

```diff
@@ -0,0 +1,128 @@
+/*
+* Licensed to the Apache Software Foundation (ASF) under one or more
+* contributor license agreements. See the NOTICE file distributed with
+* this work for additional information regarding copyright ownership.
+* The ASF licenses this file to You under the Apache License, Version 2.0
+* (the "License"); you may not use this file except in compliance with
+* the License. You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+ */
+
+package http
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/noctarius/timescaledb-event-streamer/internal/logging"
+	"github.com/noctarius/timescaledb-event-streamer/internal/sysconfig"
+	spiconfig "github.com/noctarius/timescaledb-event-streamer/spi/config"
+	"github.com/noctarius/timescaledb-event-streamer/testsupport"
+	"github.com/noctarius/timescaledb-event-streamer/testsupport/testrunner"
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"net/http"
+	"testing"
+	"time"
+)
+
+type HttpIntegrationTestSuite struct {
+	testrunner.TestRunner
+}
+
+func TestHttpIntegrationTestSuite(
+	t *testing.T,
+) {
+
+	suite.Run(t, new(HttpIntegrationTestSuite))
+}
+
+func (hits *HttpIntegrationTestSuite) Test_Http_Sink() {
+	topicPrefix := lo.RandomString(10, lo.LowerCaseLettersCharset)
+
+	httpLogger, err := logging.NewLogger("Test_Http_Sink")
+	if err != nil {
+		hits.T().Error(err)
+	}
+
+	address := "localhost:8090"
+	endpoint := "/test"
+
+	hits.RunTest(
+		func(ctx testrunner.Context) error {
+			collected := make(chan bool, 1)
+			envelopes := make([]testsupport.Envelope, 0)
+
+			http.HandleFunc(endpoint, func(w http.ResponseWriter, req *http.Request) {
+				decoder := json.NewDecoder(req.Body)
+				envelope := testsupport.Envelope{}
+				err := decoder.Decode(&envelope)
+				if err != nil {
+					httpLogger.Fatalf("Could not deserialize envelope!")
+				}
+				httpLogger.Debugf("EVENT: %+v", envelope)
+				envelopes = append(envelopes, envelope)
+				if len(envelopes) >= 10 {
+					collected <- true
+				}
+			})
+
+			go func() {
+				if err := http.ListenAndServe(address, nil); err != nil {
```

Comment:
Wouldn't it be better to use an actual instance and shut it down cleanly in the tear down? That way you also wouldn't be required to use a fixed port number (while I don't think it's a real issue though).

Reply:
It absolutely would, that was my first approach as well, but I wrote less than 10k lines of golang in total, so I couldn't figure out why my server instance seems to have stopped instantly. I could make it work this way though.

I tried my best for the second try, let me know if it's better this way or not. Thanks! :)